internal/pkg/service/appproxy/http/router.go

```diff
@@ -95,65 +96,87 @@ func (r *Router) CreateHandler() http.Handler {
 	return mux
 }
 
-func (r *Router) createDataAppHandler(app DataApp) (http.Handler, error) {
+func (r *Router) createConfigErrorHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusForbidden)
+		fmt.Fprintln(w, "Application has misconfigured OAuth2 provider.")
+	})
+}
+
+func (r *Router) createDataAppHandler(ctx context.Context, app DataApp) http.Handler {
+	ctx = ctxattr.ContextWith(ctx, attribute.String("app", string(app.ID)))
+
 	if len(app.Providers) == 0 {
-		return r.publicAppHandler(app)
+		return r.publicAppHandler(ctx, app)
 	}
-	return r.protectedAppHandler(app)
+	return r.protectedAppHandler(ctx, app)
 }
 
-func (r *Router) publicAppHandler(app DataApp) (http.Handler, error) {
+func (r *Router) publicAppHandler(ctx context.Context, app DataApp) http.Handler {
 	target, err := url.Parse("http://" + app.UpstreamHost)
 	if err != nil {
-		return nil, errors.Errorf(`cannot parse upstream url "%s" for app %s: %w`, app.UpstreamHost, app.ID, err)
+		r.logger.Errorf(ctx, `cannot parse upstream url "%s" for app <app> %s: %w`, app.UpstreamHost, app.Name, err.Error())
+		return r.configErrorHandler
```

Comment:
In future, we will need probably to include error details in the error page (at least an `exceptionId`), so the handler won't be same.

Reply:
See `httpserver/error.go` and `WithExceptionID` interfaces for example.

Error in the `go-client` implements `WithExceptionID` interface.

![image](https://github.com/keboola/keboola-as-code/assets/19371734/9685b153-9cc8-408e-a731-484dff193311)

Reply:
So you should:
- Generate an exeception ID for the error.
- Log the error, with details.
- In the handler, always return the same `exceptionID`.
- So we (support) can find it in the logs, and get error details.
- Logged message and message for the user may differ.